programs/bpf_loader/src/syscalls/sysvar.rs

```diff
@@ -160,3 +160,74 @@ declare_builtin_function!(
         )
     }
 );
+
+const SYSVAR_NOT_FOUND: u64 = 2;
+const OFFSET_LENGTH_EXCEEDS_SYSVAR: u64 = 1;
+
+// quoted language from SIMD0127
+// because this syscall can both return error codes and abort, well-ordered error checking is crucial
+declare_builtin_function!(
+    /// Get a slice of a Sysvar in-memory representation
+    SyscallGetSysvar,
+    fn rust(
+        invoke_context: &mut InvokeContext,
+        sysvar_id_addr: u64,
+        var_addr: u64,
+        offset: u64,
+        length: u64,
+        _arg5: u64,
+        memory_mapping: &mut MemoryMapping,
+    ) -> Result<u64, Error> {
+        let check_aligned = invoke_context.get_check_aligned();
+        let ComputeBudget {
+            sysvar_base_cost,
+            cpi_bytes_per_unit,
+            mem_op_base_cost,
+            ..
+        } = *invoke_context.get_compute_budget();
+
+        // Abort: "Compute budget is exceeded."
+        consume_compute_meter(
+            invoke_context,
+            sysvar_base_cost
+                .saturating_add(32_u64.div_ceil(cpi_bytes_per_unit))
+                .saturating_add(std::cmp::max(
+                    length.div_ceil(cpi_bytes_per_unit),
+                    mem_op_base_cost,
+                )),
```

Comment:
These divisions should be floor. The SIMD just uses `/` (🙃) but FD previously mentioned floor was preferred.

Reply:
updated! as for the tests, i prefered to keep them together to test that both syscall results match the same "clean" object (and thus match each other). maybe i should keep them together but use separate memory mappings tho?

Reply:
That's a good point. Maybe the tests are ok as-is. I don't feel strongly about it, so up to you!